### PR TITLE
cmd/tracee-ebpf: add capabilities dropping bypasses

### PR DESCRIPTION
**Pull request by AlonZivony**
## Initial Checklist

- [x] There is an issue describing the need for this PR.
- [x] Git log contains summary of the change.
- [x] Git log contains motivation and context of the change.
- [ ] If part of an EPIC, PR git log contains EPIC number.
- [ ] If part of an EPIC, PR was added to EPIC description.

## Description (git log)

Add options to bypass capabilities dropping partly or entirely
in tracee-ebpf.
This will help to solve locally capabilities issues for developers or
users which face capabilities bug.

Fixes: #1996

## Type of change

- [ ] Bug fix (non-breaking change fixing an issue, preferable).
- [ ] Quick fix (minor non-breaking change requiring no issue, use with care)
- [x] Code refactor (code improvement and/or code removal)
- [x] New feature (non-breaking change adding functionality).
- [ ] Breaking change (cause existing functionality not to work as expected).

## Final Checklist:

Pick "Bug Fix" or "Feature", delete the other and mark appropriate checks.

- [x] I have made corresponding changes to the documentation.
- [x] My code follows the style guidelines (C and Go) of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented all functions/methods created explaining what they do.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix, or feature, is effective.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published before.

## Git Log Checklist:

My commits logs have:

- [x] Subject starts with "subsystem|file: description".
- [x] Do not end the subject line with a period.
- [x] Limit the subject line to 50 characters.
- [x] Separate subject from body with a blank line.
- [x] Use the imperative mood in the subject line.
- [x] Wrap the body at 72 characters.
- [x] Use the body to explain what and why instead of how.

### CODE REVIEW

File: cmd/tracee-ebpf/internal/flags/capabilities.go
Comment (NDStrahilevitz) [inline cmd/tracee-ebpf/internal/flags/capabilities.go:1]: FYI: This file needs to move from internal

File: docs/deep-dive/dropping-capabilities.md
Comment (NDStrahilevitz) [inline docs/deep-dive/dropping-capabilities.md:31]:
```suggestion
This may cause a wide variety of errors.
```

File: docs/deep-dive/dropping-capabilities.md
Comment (NDStrahilevitz) [inline docs/deep-dive/dropping-capabilities.md:33]:
```suggestion
Our team tries to solve bugs as quickly as possible, but it might
```

File: docs/deep-dive/dropping-capabilities.md
Comment (NDStrahilevitz) [inline docs/deep-dive/dropping-capabilities.md:37]:
```suggestion
To fix specific missing capabilities errors locally, users can add capabilities
```

File: docs/deep-dive/dropping-capabilities.md
Comment (NDStrahilevitz) [inline docs/deep-dive/dropping-capabilities.md:41]:
```suggestion
In addition, the `--caps cancel-drop` flag can be used to cancel capability
dropping in tracee-ebpf. For docker users, the environment variable
```

**Review by NDStrahilevitz — CHANGES_REQUESTED** [review]
So I did some testing, and some UX improvements/possibly bugs that I think should be addressed:
1. `--caps add <cap_name>` doesn't work, neither does `--caps add=<cap_name>`
2. With correct usage (`--caps <cap_name>`) - giving a non real cap name (tried with `CAP_FAKE`) doesn't return any error, instead continuing execution, I think that's bad UX for a user for example if a typo is made.

**Comment by NDStrahilevitz** [comment]
> So I did some testing, found two UX improvements/possibly bugs that I think should be addressed:
> 
> 1. `--caps add <cap_name>` doesn't work, neither does `--caps add=<cap_name>`
> 2. With correct usage (`--caps <cap_name>`) - giving a non real cap name (tried with `CAP_FAKE`) doesn't return any error, instead continuing execution, I think that's bad UX for a user for example if a typo is made.

Was this also addressed? If so I can check and approve.

File: cmd/tracee-ebpf/flags/capabilities.go
Comment (NDStrahilevitz) [inline cmd/tracee-ebpf/flags/capabilities.go:88]: Ok, just checked this and it works but only for lower case capability names. 
So `cap_sys_admin` works but `CAP_SYS_ADMIN` doesn't, I assume library uses only lower case names for some reason. 
But in the [man page](https://man7.org/linux/man-pages/man7/capabilities.7.html) capabilities are upper case so they should probably also be supported.

**Review by NDStrahilevitz — CHANGES_REQUESTED** [review]
Ok, so there's a ux "bug" with the capability names, other than that all good so I can approve right after.

File: cmd/tracee-ebpf/flags/capabilities.go
Comment (NDStrahilevitz) [inline cmd/tracee-ebpf/flags/capabilities.go:44]: Do we care here about any operators besides`=` and `!`? Also, docs don't allow for `!=` so either that logic (and operator) should be removed or it should be documented.

**Review by NDStrahilevitz — CHANGES_REQUESTED** [review]
One final catch, sorry for not noticing before.

File: cmd/tracee-ebpf/capabilities.go
Comment (NDStrahilevitz) [inline cmd/tracee-ebpf/capabilities.go:11]: nit: this moved downwards but it should be before the github imports

**Comment by NDStrahilevitz** [comment]
Please rebase this so all matrix tests can run.

File: cmd/tracee-ebpf/flags/capabilities.go
Comment (NDStrahilevitz) [inline cmd/tracee-ebpf/flags/capabilities.go:32]: In most our examples we just have `--flagname input input....` no need to have ./tracee-ebpf here

**Review by NDStrahilevitz — COMMENTED** [review]
Final comment on the flag's usage examples, otherwise lgtm

**Review by NDStrahilevitz — CHANGES_REQUESTED** [review]
Commit title and body got switched somehow

**Comment by rafaeldtinoco** [comment]
@AlonZivony I believe https://github.com/aquasecurity/tracee/pull/2033 should be merged first and then you rebase (as that one is simpler) and then once this one is good we merge this one.

**Comment by AlonZivony** [comment]
> @AlonZivony I believe #2033 should be merged first and then you rebase (as that one is simpler) and then once this one is good we merge this one.

Np, but we want it to enter to the coming sub-release so it depends if this PR will enter soon or not too.

**Review by NDStrahilevitz — APPROVED** [review]
LGTM!

**Comment by rafaeldtinoco** [comment]
NOTE: We're waiting 0.8.1 release first before merging this one.

**Comment by NDStrahilevitz** [comment]
We're into 0.8.2 cycle so this can be merged. @yanivagman or @rafaeldtinoco do you want to go over first?

**Review by yanivagman — APPROVED** [review]
LGTM

**Pull request merged**
